webpack.hot.config.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Webpack development server configuration
+ */
+
+'use strict';
+
+var webpack = require('webpack');
+var baseConfig = require('./webpack.config.js');
+
+module.exports = Object.assign({}, baseConfig, {
+  debug: true,
+
+  entry: [
+    'webpack/hot/only-dev-server'
+  ].concat(baseConfig.entry),
+
+  plugins: baseConfig.plugins.concat([
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin()
+  ])
+});
+
+module.exports.module.loaders[0].loader = 'react-hot!babel-loader';
```

Comment:
I think something like this is more readable and easy to understand because it reflects your idea of aggregating config (you add one loader to the previous ones at the beginning of the list)

```js
loaders: baseConfig.module.loaders.unshift({
  test: /\.js$/,
  exclude: /node_modules/,
  loader: 'react-hot'
})
```